apps/next-evans/lib/types.ts

```diff
@@ -39,26 +41,29 @@ export interface DocumentType {
   numDownloads: number;
   customCSS: string | null;
   customHTML: string | null;
-  dateAdded: string;
+  createdAt: ISODateString;
   isbn: string;
 }
 
-export type DocumentWithMeta = DocumentType & {
+export type Doc<T extends keyof Document = never> = Pick<
+  Document,
+  'isbn' | 'title' | 'slug' | 'customCSS' | 'customHTML' | 'authorName' | 'authorSlug' | T
+>;
```

Comment:
study this type for a minute -- it's the utility generic that allows us to grab a bunch of common props that _many_ components need, and only specify the extras. when we use this on new components in the future, we should be careful that we truly are using all of the props listed on this line.

TS note: by setting the generic to have a default of `never`, it allows us to just use the type as `Doc` in places where all we want are the props already listed here. because `T | never` is the same as `T`. 🤓